src/loader/x86_64/elf/mod.rs

```diff
@@ -343,15 +341,26 @@ where
         }
 
         // Skip the note header plus the size of its fields (with alignment).
-        let namesz_aligned = align_up(u64::from(nhdr.n_namesz), n_align)?;
-        let descsz_aligned = align_up(u64::from(nhdr.n_descsz), n_align)?;
+        let namesz_aligned = align_up(u64::from(nhdr.n_namesz), phdr.p_align)?;
+        let descsz_aligned = align_up(u64::from(nhdr.n_descsz), phdr.p_align)?;
+
+        // `namesz` and `descsz` are both `u32`s. They refer also to 4 byte sized and aligned memory
```

Comment:
The commnet also needs to be updated as it is not 4-bytes.

Reply:
Done.